wallet/wallet-node-client/src/handles_client/mod.rs

```diff
@@ -147,7 +147,11 @@ impl NodeInterface for WalletHandlesClient {
     }
 
     async fn submit_transaction(&self, tx: SignedTransaction) -> Result<(), Self::Error> {
-        self.p2p.call_async_mut(move |this| this.submit_transaction(tx)).await??;
+        // TODO: Should this return mempool::TxStatus?
```

Comment:
This should return the low-level result of the call, whether TxStatus or otherwise.

The wallet controller then should interpret that result and inform the user about it in the appropriate manner, which may be GUI or CLI.

Reply:
> This should return the low-level result of the call, whether TxStatus or otherwise.

That is currently not that completely trivial to achieve without pulling some heavy dependencies into the wallet interface. The most straightforwad way to achieve it is to split out the `TxStatus` type out of mempool (like introducing `mempool-types` or such). That looks like a job for a separate PR.

Reply:
I see. Though `mempool::TxStatus` is quite a simple type. If we need a mempool-types, I don't see a problem with that. You can make it in another PR on top of this one. Keep in mind that without that, there's no way for the wallet to know what happened with the transactions.